engine/runtime-parser/src/test/java/org/enso/compiler/core/IrPersistanceTest.java

```diff
@@ -61,6 +67,30 @@ public void scalaMap() throws Exception {
     assertEquals(in, out);
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  public void scalaImmutableMapIsLazy() throws Exception {
+    var s1 = new LazySeq("Hello");
+    var s2 = new LazySeq("World");
+    var in = (scala.collection.immutable.Map) scala.collection.immutable.Map$.MODULE$.empty()
+      .$plus(new Tuple2("Hello", s1))
+      .$plus(new Tuple2("World", s2));
+
+    LazySeq.forbidden = true;
+    var out = (scala.collection.immutable.Map)
+      serde(scala.collection.immutable.Map.class, in, 64);
```

Comment:
The Java version below works. The Scala version fails with following exception:
```
sbt:runtime-parser> testOnly *IrPersistanceTest
 Test org.enso.compiler.core.IrPersistanceTest.scalaMapIsImmutable failed: java.lang.IllegalStateException: Cannot create LazySeq right now!, took 0.003 sec
     at org.enso.compiler.core.IrPersistanceTest$LazySeq.<init>(IrPersistanceTest.java:336)
     at org.enso.compiler.core.IrPersistanceTest$PersistLazySeq.readObject(IrPersistanceTest.java:391)
     at org.enso.compiler.core.IrPersistanceTest$PersistLazySeq.readObject(IrPersistanceTest.java:376)
     at org.enso.persist.Persistance.readWith(Persistance.java:144)
     at org.enso.persist.PerInputImpl.readInline(PerInputImpl.java:45)
     at org.enso.persist.PerBufferReference.readObject(PerBufferReference.java:27)
     at org.enso.persist.Persistance$Reference.get(Persistance.java:205)
     at org.enso.compiler.core.ir.IrLazyMap$En.getValue(IrLazyMap.java:59)
     at scala.collection.convert.JavaCollectionWrappers$JMapWrapperLike$$anon$5.next(JavaCollectionWrappers.scala:400)
     at scala.collection.convert.JavaCollectionWrappers$JMapWrapperLike$$anon$5.next(JavaCollectionWrappers.scala:397)
     at scala.collection.mutable.Growable.addAll(Growable.scala:62)
     at scala.collection.mutable.Growable.addAll$(Growable.scala:57)
     at scala.collection.immutable.MapBuilderImpl.addAll(Map.scala:690)
     at scala.collection.immutable.Map$.from(Map.scala:641)
     at scala.collection.immutable.Map.from(Map.scala)
     at org.enso.compiler.core.ir.IrPersistance$PersistScalaMap.readObject(IrPersistance.java:237)
     at org.enso.compiler.core.ir.IrPersistance$PersistScalaMap.readObject(IrPersistance.java:212)
     at org.enso.persist.Persistance.readWith(Persistance.java:144)
     at org.enso.persist.PerInputImpl.readInline(PerInputImpl.java:45)
     at org.enso.persist.PerBufferReference.readObject(PerBufferReference.java:27)
     at org.enso.persist.Persistance$Reference.get(Persistance.java:205)
     at org.enso.compiler.core.IrPersistanceTest.serde(IrPersistanceTest.java:315)
     at org.enso.compiler.core.IrPersistanceTest.scalaMapIsImmutable(IrPersistanceTest.java:81)
     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
     at java.lang.reflect.Method.invoke(Method.java:580)
```
What is the best way to convert `java.util.Map` to `scala.collections.immutable.Map`? @4e6, @hubertp?

Reply:
But you cannot convert collections between Java and Scala without copying anyway

Reply:
https://github.com/enso-org/enso/pull/8359/commits/f9c34a94dca334411f9565eca4cac9ef3fb5778d